layers/+lang/groovy/config.el

```diff
@@ -0,0 +1,20 @@
+;;; config.el --- Groovy layer configuration file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Johnson Denen <johnson.denen@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defun groovy/gradle-clean-build ()
```

Comment:
These functions should not be in `config.el` file. I think the best place is `funcs.el`([doc][1]). Also it would be good to 'guard' these functions by placing them inside of `(when (configuration-layer/package-usedp 'gradle-mode) FUNCS)`, so they don't error if someone calls them with `gradle-mode` excluded. 

  [1]: https://github.com/syl20bnr/spacemacs/blob/master/doc/LAYERS.org#funcsel